pip_audit/_format/junit.py

```diff
@@ -0,0 +1,210 @@
+"""
+Functionality for formatting vulnerability results as an array of JSON objects.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import cast
+
+import pip_audit._fix as fix
+import pip_audit._service as service
+
+from .interface import VulnerabilityFormat
+
+
+class JunitFormat(VulnerabilityFormat):
+    """
+    An implementation of `VulnerabilityFormat` that formats vulnerability results as a
+    junit xml file (https://github.com/testmoapp/junitxml).
+    Junit is a wellknow and usefull format when working with Continuous Integration.
+    """
+
+    def __init__(self, output_desc: bool, output_aliases: bool):
+        """
+        Create a new `JunitFormat`.
+
+        `output_desc` is a flag to determine whether descriptions for each vulnerability should be
+        included in the output as they can be quite long and make the output difficult to read.
+
+        `output_aliases` is a flag to determine whether aliases (such as CVEs) for each
+        vulnerability should be included in the output.
+        """
+        self.output_desc = output_desc
+        self.output_aliases = output_aliases
+
+    @property
+    def is_manifest(self) -> bool:
+        """
+        See `VulnerabilityFormat.is_manifest`.
+        """
+        return False
+
+    def format(
+        self,
+        result: dict[service.Dependency, list[service.VulnerabilityResult]],
+        fixes: list[fix.FixVersion],
+    ) -> str:
+        """
+        Returns a Junit formatted string for a given mapping of dependencies to vulnerability
+        results.
+
+        See `VulnerabilityFormat.format`.
+        """
+        nb_tests = 0
+        nb_failures = 0
+        nb_skipped = 0
+        nb_errored = 0
+        timestamp = datetime.now().isoformat()
+
+        output = ''
+        for dep, vulns in result.items():
+            te, sk, fa, er, outp = self._format_dep(dep, vulns, fixes)
+            nb_tests += te
+            nb_skipped += sk
+            nb_failures += fa
+            nb_errored += er
+            output += outp
+        output_xml = '<?xml version="1.0" encoding="UTF-8"?>\n'
+        output_xml += f'<testsuites name="pip-audit" tests="{nb_tests}" failures="{nb_failures}" '
+        output_xml += f'skipped="{nb_skipped}" errors="{nb_errored}" timestamp="{timestamp}">\n'
+        output_xml += output
+        output_xml += '</testsuites>'
+        return output_xml
```

Comment:
I'm sorry, but this approach is a hard blocker: we can't be manually building up XML with string concatenation, *especially* without any proper escaping.